test/libsolidity/GasMeter.cpp

```diff
@@ -176,8 +176,8 @@ BOOST_AUTO_TEST_CASE(store_keccak256)
 	char const* sourceCode = R"(
 		contract test {
 			bytes32 public shaValue;
-			constructor(uint a) public {
-				shaValue = keccak256(abi.encodePacked(a));
+			constructor() public {
+				shaValue = keccak256(abi.encodePacked(this));
```

Comment:
I'm not quite sure why this is needed and why it works, actually...

Reply:
The test input did not provide the constructor argument. Previously, it worked because something else was used instead... Now we have the check.

The parameter to `keccak256` has to be dynamic, otherwise the optimizer will just replace it by a constant.